### ISSUE TITLE
TwitchLib Unity - Helix requests are unresponsive 

### ISSUE THREAD

**Issue by turkenberg** [TwitchLib/TwitchLib#1041]
``Hi,
Been mingling around with TwitchLib and Unity for 6 months now.
PubSub and V5 works fine, same for client, but Helix seems to face an authentication issue.

### Flow
Simply create a new Unity project, import TwitchLib DLLs and add the example script on any scene object.
Enter your client ID and your oauth token in the provided example script
Hit play in Unity Editor
Press Alpha1, then press Alpha2.

### Expected
1 returns the list of videos in console from user using its user_id--> OK
2 returns the list of videos in console  from user using its username ; or returns a log error if something goes wrong --> KO

### Result
1 returns the list, that's correct
2 actually returns nothing. The async call remains hanging and no other information is provided.

### Notes
1. If you don't provide an oauth token, the request actually returns an error message coming from Twitch. But as soon as the token is provided, it responds nothing (especially if I use random string as the token).
2. This actually verifies for all Helix calls made from Unity framework using TwitchLib. I have never been able to make one :)
3. The example in README is outdated ; library structure is modified a bit. This is why I included the example script, slightly modified.

[TwitchLibUnityExample.zip](https://github.com/TwitchLib/TwitchLib/files/6562770/TwitchLibUnityExample.zip)

If this is an error on my side I am terribly sorry.

**Comment by swiftyspiffy** [comment]
So, i don't know a ton about how Unity works. In fact, I know very little. I can provide you an example of something I used to test, as well as a VOD of a Twitch streamer using the library in her project.

My example:
https://gist.github.com/swiftyspiffy/08813a38f6bcf472251e4379117bfd96

VOD example from ProfoundRice:
https://www.twitch.tv/videos/1026779639?t=1h5m21s

**Comment by turkenberg** [comment]
Hi Swifty,

Thanks for those ;
I suspected credentials to be wrong on my side. They were!
maybe a good addon would be a simple log informing Unity about 401 / unauthorized or something ?

The part I totally missed was getting the client_id and token from id.twitch.tv etc. etc.
So I used your awesome tool again and with those it works like a charm.

One very quick question regarding your tool:
1. Why my bot account and my own account have the same client_id when using your (awesome) tool ?
is it the client_id of your tool ?

Again this library is awesome, a big thanks to everyone who made it and maintains it.

Best,
Turk
